integration_tests/suite/test_cdr.py

```diff
@@ -214,10 +202,70 @@ def test_given_id_when_get_cdr_by_id_csv_then_get_cdr_by_id_csv(self):
                 source_internal_extension='5938',
                 source_internal_context='internal',
                 source_user_uuid=USER_1_UUID,
+                source_tenant_uuid=MASTER_TENANT,
                 tags=any_of('rh;Poudlard', 'Poudlard;rh'),
             )
         )
 
+    @call_logs([
+        {'id': 10,
+         'date': '2017-03-23 00:00:00',
+         'participants': [{'user_uuid': USER_1_UUID,
+                           'tenant_uuid': USERS_TENANT,
+                           'line_id': '1',
+                           'role': 'source'}]},
+        {'id': 11,
+         'date': '2017-03-23 00:00:00',
+         'participants': [{'user_uuid': USER_2_UUID,
+                           'tenant_uuid': USERS_TENANT,
+                           'line_id': '1',
+                           'role': 'source'}]},
+        {'id': 12,
+         'date': '2017-03-23 00:00:00',
+         'participants': [{'user_uuid': OTHER_USER_UUID,
+                           'tenant_uuid': OTHER_TENANT,
+                           'line_id': '1',
+                           'role': 'source'}]}
+    ])
+    def test_get_cdr_by_id_multitenant(self):
+        self.call_logd.set_token(USER_1_TOKEN)
+        result = self.call_logd.cdr.get_by_id(10)
+        assert_that(result, has_entries(source_user_uuid=USER_1_UUID,
+                                        source_tenant_uuid=USERS_TENANT))
+
+        result = self.call_logd.cdr.get_by_id(11)
+        assert_that(result, has_entries(source_user_uuid=USER_2_UUID,
+                                        source_tenant_uuid=USERS_TENANT))
+
+        assert_that(
+            calling(self.call_logd.cdr.get_by_id).with_args(cdr_id=12),
+            raises(CallLogdError).matching(
+                has_properties(
+                    status_code=404,
+                    message=contains_string_ignoring_case('no cdr found'),
+                    details=has_key('cdr_id')
+                )
+            )
+        )
+
+        self.call_logd.set_token(OTHER_USER_TOKEN)
+        result = self.call_logd.cdr.get_by_id(12)
+        assert_that(result, has_entries(source_user_uuid=OTHER_USER_UUID,
+                                        source_tenant_uuid=OTHER_TENANT))
```

Comment:
I first check to get cdr `12` with `USER_1_TOKEN` it returns 404, it could mean, it really doesn't exists or it exists but its not belong my tenant. So I check with OTHER_USER_TOKEN to ensure its the seconds hypotheses.

Reply:
Is the only one that I can understand. But I have to say that we must trust our precondition (otherwise the test is invalid), so we must trust that call log exists